### PR TITLE
[ntuple] add ClassDef to RKeyBlob

### PR DESCRIPTION
**Pull request by silverweed**
Same as https://github.com/root-project/root/commit/19caf55b88d16ac67d9bfb39380a4d1c89f935e6 but this time moving RKeyBlob out of the anon namespace which causes compile errors on windows (I missed this in the previous commit, now reverted).

# This Pull request:

## Checklist:

- [x] tested changes locally
- [ ] updated the docs (if necessary)

This PR fixes #

### CODE REVIEW

File: tree/ntuple/v7/src/RMiniFile.cxx [hunk 1]
```diff
@@ -971,9 +978,10 @@ class RKeyBlob : public TKey {
       Create(nbytes);
       *seekKey = fSeekKey;
    }
+
+   ClassDefInline(RKeyBlob, 0)
```
Comment (silverweed) [inline tree/ntuple/v7/src/RMiniFile.cxx:982]: I put version=0 because in my understanding this class never gets serialized by the ROOT I/O

Comment (vepadulano) [inline tree/ntuple/v7/src/RMiniFile.cxx:982]:
```suggestion
   ClassDefInlineOverride(RKeyBlob, 0)
```

File: tree/ntuple/v7/src/RMiniFile.cxx [hunk 2]
```diff
@@ -954,10 +955,16 @@ constexpr char const *kBlobClassName = "RBlob";
 /// The class name of the RNTuple anchor
 constexpr char const *kNTupleClassName = "ROOT::Experimental::RNTuple";
 
+} // anonymous namespace
+
 /// The RKeyBlob writes an invisible key into a TFile.  That is, a key that is not indexed in the list of keys,
 /// like a TBasket.
+/// NOTE: out of anonymous namespace because otherwise ClassDefInline fails to compile
+/// on some platforms.
 class RKeyBlob : public TKey {
```
Comment (jblomer) [inline tree/ntuple/v7/src/RMiniFile.cxx:964]: This becomes now a global symbol name, doesn't it? In any case, should we put it in `ROOT::Experimental::Internal`?

Comment (pcanal) [reply tree/ntuple/v7/src/RMiniFile.cxx:964]: Yes and yes :)

An anonymous namespace is usually implemented by an untypable name .... which by definition wont be type-able by the `ClassDef` macro or dictionary generation.

**Pull request merged**